api/database/migrations/2023_09_20_211150_combine_pool_skills.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        // create the combined table
+        Schema::create('pools_skills', function (Blueprint $table) {
+            $table->uuid('skill_id');
+            $table->foreign('skill_id')->references('id')->on('skills');
+            $table->uuid('pool_id');
+            $table->foreign('pool_id')->references('id')->on('pools');
+            $table->unique(['skill_id', 'pool_id']);
```

Comment:
The good news is that this PR also fixes #4302.  The bad news is that #4302 means that this migration might fail in prod on deployment.  I suggest either confirming that it won't fail in prod ahead of time or updating this migration to gracefully handle it. Maybe by collecting both batches of skills and deduplicating them first? I guess if a skill is marked as both then it should go in as essential.

Reply:
@vd1992 is already ahead of us, @petertgiles! 😆 Deduplication happens at line 37: 
```
$nonessentialPoolSkills = DB::table('pools_nonessential_skills')
            ->whereNotIn('skill_id', $essentialPoolSkillsIds) // do not have repeat pool-skill entities with different type
            ->get();
```

Reply:
🤦‍♂️ 

Unresolving because looking more closely at that code block, I need to update the filtering. 
It needs to filter for unique `skill_id` AND `pool_id`, as right now I think it just makes it so you can't have any skill be used as an asset and essential for any pool. Which is bad (only unique `skill_id` on the pivot)

Reply:
Okay, now I think it should work as intended, testing and it migrates successfully while removing the actual duplicates

https://github.com/GCTC-NTGC/gc-digital-talent/pull/7953/commits/9609b5dc312a3e2de1f7ea9c4b292ac020eacdd2

Reply:
:+1: